features/support/stub_feedback_requests.rb

```diff
@@ -2,30 +2,42 @@
 
 Before('@stub_feedback_success') do
   stub_request(:post, %r{\A#{feedback_service_url}} )
-    .and_return(status: 201, body: successful_zendesk_body.to_json)
+    .and_return(status: 201, body: successful_feedback_body.to_json)
 end
 
 Before('@stub_feedback_failure') do
   stub_request(:post, %r{\A#{feedback_service_url}} )
-    .and_return(status: 500, body: unsuccessful_zendesk_body.to_json)
+    .and_return(status: 500, body: unsuccessful_feedback_body.to_json)
 end
 
 Before('@stub_bug_report_success') do
   stub_request(:post, %r{\Ahttps://ministryofjustice.zendesk.com/api/v2/.*} )
-  .and_return(status: 201, body: successful_zendesk_body.to_json)
+  .and_return(status: 201, body: { item: {} }.to_json)
 end
 
 Before('@stub_bug_report_failure') do
   stub_request(:post, %r{\Ahttps://ministryofjustice.zendesk.com/api/v2/.*} )
-    .and_return(status: 500, body: unsuccessful_zendesk_body.to_json)
+    .and_return(status: 500, body: { error: 'Unsuccessful' }.to_json)
 end
 
-def successful_zendesk_body
-  { item: {} }.to_json
+def successful_feedback_body
+  if Settings.zendesk_feedback_enabled?
+    # Zendesk
+    { item: {} }.to_json
+  else
+    # Survey Monkey
+    { id: '123' }.to_json
+  end
```

Comment:
For the sake of clarity I would suggest going back to `successful_zendesk_body` (etc) and creating a new `successful_survey_monkey_body` function. Whilst what you have here is good for production code, it is better to have clarity rather than efficiency in test code.

Potentially there could be a new test for changing the `zendesk_feedback_enabled` setting.